pkg/operator/controllers/dnsmasq/machineconfig_controller.go

```diff
@@ -71,8 +71,11 @@ func (r *MachineConfigReconciler) Reconcile(ctx context.Context, request ctrl.Re
 		r.log.Error(err)
 		return reconcile.Result{}, err
 	}
+	if mcp.GetDeletionTimestamp() != nil {
```

Comment:
TODO: remove this as it is no longer necessary (remnant of old implementation via finalizers).